test/cbmc/proofs/handleSelfTestJobDoc/handleSelfTestJobDoc_harness.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * AWS IoT Over-the-air Update v3.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file handleSelfTestJobDoc_harness.c
+ * @brief Implements the proof harness for handleSelfTestJobDoc function.
+ */
+/*  Ota Agent includes. */
+#include "ota.h"
+#include "stubs.h"
+
+extern OtaAgentContext_t otaAgent;
+extern void handleSelfTestJobDoc( const OtaFileContext_t * pFileContext );
+
+OtaErr_t validateUpdateVersion( const OtaFileContext_t * pFileContext )
+{
```

Comment:
I don't think this override is necessary. 
handleSelfTestJobDoc() should validate it's parameter via an assertion.

Reply:
@paulbartell In this case an assertion to validate the parameter is not required since the test function is an internal function and will be called with a statically allocated variable. The override was required because if there is no body for validateUpdateVersion then CBMC would assume `OtaErr_t` as an `unsigned int` and return values which are out of the values from the enum.

Reply:
So could a solar flare. That's kinda the point.

Reply:
I am fine either way - have a stub or don't have a stub. As long as this function is checked in another proof, I am happy.
It is up to you @paulbartell :)

Reply:
Is there a particular reason to constrain the return value of validateUpdateVersion? It looks like the current implementation of handleSelfTestJobDoc could handle any value.

Reply:
No particular reason. If we havoc the `otaAgent`, which we should, then we can call the function in the library. And since the call-tree is not huge, we can get away with this (if it were huge, we'd need to prune some branches by stubbing them out or else the proofs would take too much time to complete). Otherwise, we'd need a stub to return a range of values.